### ISSUE TITLE
UtyMap

### ISSUE THREAD

**Issue by zhaishengfu** [RaceTheMaSe/UtyMapUnreal#2]

**Comment by RaceTheMaSe** [comment]
There is no dependency on Unity at all. The Unreal part depends on UE4RuntimeMeshComponent from Koderz with a small addition from me and building the core library from UtyMap yourself, which is itself depends on CMake, Boost 1.62, protobuf 2.6.1 and zlib to build.

But for convenience I put a compiled UtyMap.Shared.dll (and .lib) for Unreal up at [Link](https://drive.google.com/open?id=0Bw6cJNSOahkWQmxzX19wSUJQbDA). That should help you skip the compilation of UtyMap. Still you need to check it out for the Unreal project to build.

I noticed a small typo error: Change FRuntimeMeshVertexTripleUV to FRuntimeMeshVertexTrippleUV in DemoModelBuilder.cpp to compile the Unreal project or vice versa so all occurances are named the same.

Please let me know if you still face issues compiling.

**Comment by zhaishengfu** [comment]
Thank you very much. Yes I noticed the error of 'FRuntimeMeshVertexTripleUV' and 'FRuntimeMeshVertexTrippleUV', and indeed i compiled all the sub components right(cmake, boost, protobuf and zlib).but when i compiled the utymap core component, it says ' x86 and x64 incompatible' error. I guess this is because I compiled win32 somewhere. But After I Compiled using all x64 platform, It Still has this error. I guess this is the error of zlib. I will try later time.  Any way, Thank you very much for your compiled version. Good luck

**Comment by RaceTheMaSe** [comment]
Yes, you are right, its VS2015 x64 Release. UtymapUnreal is a UE 4.14.3 project but you can use 4.15 as well, if you adjust the .uproject file.

If you want to change a mesh, look into MapTileAdapter. This is where meshes from the core library are received. But that would contradict this approach to encapsulate generation into an independet library. To exclude automatic mesh generation for certain buildings, look in the originals repo's [Wiki](https://github.com/reinterpretcat/utymap/wiki/How-To) under "Customize specific buildings". I have no experience with that at all.

**Comment by lonnibesancon** [comment]
@RaceTheMaSe so this only works under Windows right? I am currently working on a mac and would love to make it work on it... Also, 
https://github.com/RaceTheMaSe/UE4RuntimeMeshComponent.git is not found
